packages/file_selector/file_selector_ios/ios/Classes/FTLFileSelectorPlugin.m

```diff
@@ -0,0 +1,88 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FTLFileSelectorPlugin.h"
+
+@interface FTLFileSelectorPlugin () <UIDocumentPickerDelegate>
+
+@property(nonatomic) FlutterResult pendingResult;
+
+@end
+
+@implementation FTLFileSelectorPlugin
+
+- (void)openPickerWithCall:(FlutterMethodCall *)call
+                    result:(FlutterResult)result
+       allowMultiSelection:(BOOL)allowMultiSelection {
+  NSDictionary *acceptedTypes = call.arguments[@"acceptedTypes"];
+  UIDocumentPickerViewController *documentPicker =
+      [[UIDocumentPickerViewController alloc] initWithDocumentTypes:acceptedTypes[@"UTIs"]
+                                                             inMode:UIDocumentPickerModeImport];
+  documentPicker.delegate = self;
+  if (@available(iOS 11.0, *)) {
+    documentPicker.allowsMultipleSelection = allowMultiSelection;
+  }
+
+  UIViewController *rootVC = UIApplication.sharedApplication.delegate.window.rootViewController;
+  if (rootVC) {
+    [rootVC presentViewController:documentPicker animated:YES completion:nil];
+    self.pendingResult = result;
+  } else {
+    result([FlutterError errorWithCode:@"error"
+                               message:@"Missing root view controller."
+                               details:nil]);
+  }
+}
+
+#pragma mark - FlutterPlugin
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+  FlutterMethodChannel *channel =
+      [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.io/file_selector_ios"
+                                  binaryMessenger:registrar.messenger];
+  FTLFileSelectorPlugin *plugin = [[FTLFileSelectorPlugin alloc] init];
+  [registrar addMethodCallDelegate:plugin channel:channel];
+}
+
+- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
+  if ([@"openFile" isEqualToString:call.method]) {
+    [self openPickerWithCall:call result:result allowMultiSelection:NO];
+  } else if ([@"openFiles" isEqualToString:call.method]) {
+    [self openPickerWithCall:call result:result allowMultiSelection:YES];
```

Comment:
The platform channel layer doesn't need to directly mirror the platform interface. While you'll see that in many of our plugins, it's almost entirely for historical reasons, and it's not something we want to do in new plugins when it doesn't actually make sense. It's pretty clear here that a single method call with a boolean argument would be a much better representation; you can do that mapping in the Dart code (which is easier to test/debug in general) instead of native code.

Reply:
Done.